jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/LockFile.java

```diff
@@ -73,10 +73,16 @@ public void close() {
 
     } finally {
       try {
-        Preconditions.checkNotNull(lockMap.get(lockFile)).unlock();
         Files.delete(lockFile);
 
-      } catch (IllegalMonitorStateException | IOException ignored) {
+      } catch (IOException ignored) {
+
+      } finally {
+        try {
+          Preconditions.checkNotNull(lockMap.get(lockFile)).unlock();
+
+        } catch (IllegalMonitorStateException ignored) {
```

Comment:
Ah, I think I was wrong with catching this, sorry. I mistakenly assumed that `LockFile.close()` will be called if `ReentrantLock.lockInterruptibly()` fails in the static `LockFile.lock()` constructor. The `LockFile` instance is never instantiated in that case, so I think `IllegalMonitorStateException` can never happen. Agreed?